recipes-core/glibc-locale-tests/glibc-locale-tests/test_locale_aliases.c

```diff
@@ -0,0 +1,107 @@
+#include <locale.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <wchar.h>
+
+// L1 Locale (Equivalent to ISO-8859-1 charmap)
+// In C, you cannot use universal notation (\uXXXX) for characters in the basic character set (u0000-u009F),
+// so use normal hex here instead.
+const wchar_t* latin_unicode_string = L"\x4B\u00F6\x6E\x65\x6E\x20\u00D8\u00DA\x45"; // Könen ØÚE
+const char* latin_multibyte_file = "L1.txt";
+
+// CP932 Locale (Equivalent to WINDOWS-31J charmap)
+const wchar_t* japanese_unicode_string = L"\u540D\u524D\u306F\u306A\u3093\u3067\u3059\u304B"; // 名前はなんですか
+const char* japanese_multibyte_file = "CP932.txt";
+
+// CP936 Locale (Equivalent to GBK charmap)
+const wchar_t* chinese_unicode_string = L"\u4F60\u53EB\u4EC0\u4E48\u540D\u5B57"; // 你叫什么名字
+const char* chinese_multibyte_file = "CP936.txt";
+
+int set_locale(char* locale_alias_name) {
+    setlocale(LC_ALL, locale_alias_name);
+    return strcmp(locale_alias_name, setlocale(LC_ALL, NULL));
+}
+
+int convert_ws_to_mbs(const wchar_t* wide_char_string, char** actual_hex_string) {
+    int len = wcstombs(NULL, wide_char_string, 0);
+    if (len > 0) {
+        *actual_hex_string = malloc((len + 1) * sizeof(char));
+        return wcstombs(*actual_hex_string, wide_char_string, len + 1);
+    }
+    return len;
+}
+
+int convert_mbs_to_ws(const char* multibyte_string, wchar_t** wide_string) {
+    int len = mbstowcs(NULL, multibyte_string, 0);
+    if (len > 0) {
+        *wide_string = malloc((len + 1) * sizeof(wchar_t));
+        return mbstowcs(*wide_string, multibyte_string, len + 1);
+    }
+    return len;
+}
+
+int test_locale_conversion(char* locale_alias_name, const char* multibyte_file, const wchar_t* unicode_string) {
+    FILE* fp;
+    char multibyte_string[50];
+    wchar_t* converted_unicode_string;
+    char* converted_multibyte_string;
```

Comment:
nit: There is still a subtle bug here, where convert_ws_to_mbs and convert_mbs_to_ws can return a value <= 0 but not store a new pointer value in converted_unicode_string or converted_multibyte_string. In that case, those variables end up with a random value and I think will crash your program when you try to free them at the bottom of this function. You can avoid this by initializing them to NULL.

For clarity, this happens if the initial call to wcstombs/mbstowcs in convert_ws_to_mbs/convert_mbs_to_ws returns a len <=0.

Reply:
Ah, good call. Fixed with the initialization to NULL.